### PR TITLE
[Enhancement] support cast array as string

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #issue

Support `cast array<any> as string`.

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/exprs/cast_expr_array.cpp [hunk 1]
```diff
@@ -292,4 +295,75 @@ StatusOr<ColumnPtr> CastJsonToArray::evaluate_checked(ExprContext* context, Chun
...
+
+    size_t size = input_chunk->num_rows();
+    ArrayColumn::Ptr array_column;
+    ColumnPtr array_elements;
+
+    if (column->is_nullable()) {
+        NullableColumn::Ptr nullable = ColumnHelper::as_column<NullableColumn>(column);
+        array_column = ColumnHelper::cast_to<TYPE_ARRAY>(nullable->data_column());
+        array_elements = array_column->elements_column();
+    } else {
```
Comment (Seaven) [inline be/src/exprs/cast_expr_array.cpp:312]: check is_const

File: be/src/exprs/cast_expr_array.cpp [hunk 2]
```diff
@@ -292,4 +295,75 @@ StatusOr<ColumnPtr> CastJsonToArray::evaluate_checked(ExprContext* context, Chun
...
+                    row += ',';
+                }
+                if (nullable_array_string->is_null(j)) {
+                    row += "NULL";
+                } else {
+                    auto slice = array_string->get_slice(j);
+                    StringPiece piece(slice.data, slice.size);
+                    row += '"';
+                    strings::BackslashEscape(piece, "\"", &row);
+                    row += '"';
```
Comment (Seaven) [inline be/src/exprs/cast_expr_array.cpp]: cast array to string need cast the element to string?

File: test/sql/test_array_fn/R/test_array_cast [hunk 3]
```diff
@@ -0,0 +1,156 @@
+-- name: test_array_cast
+select cast(array<string>['a', 'b'] as string);
```
Comment (fzhedu) [inline test/sql/test_array_fn/R/test_array_cast:2]: what about empty array []?

File: test/sql/test_array_fn/R/test_array_cast [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+[NULL,[NULL]]
+-- !result
+CREATE TABLE array_test ( 
+    pk bigint not null ,
+    c1   Array<String>, 
+    c2   Array<BigInt>,
+    c3   Array<Double>,
+    c4   Array<DECIMAL(26, 2)>,
+    c5   Array<DECIMAL64(4, 3)>,
+    c6   Array<DECIMAL128(25, 19)>,
```
Comment (fzhedu) [inline test/sql/test_array_fn/R/test_array_cast:45]: add some no-supported cases?
array<map<int,int>> or array<struct<>>>?

File: be/src/exprs/cast_expr_array.cpp (hunk 2 above)
Comment (mofeiatwork) [reply be/src/exprs/cast_expr_array.cpp]: removed

File: test/sql/test_array_fn/R/test_array_cast (hunk 3 above)
Comment (mofeiatwork) [reply test/sql/test_array_fn/R/test_array_cast:2]: it's `[]`. Added test case

File: test/sql/test_array_fn/R/test_array_cast (hunk 4 above)
Comment (mofeiatwork) [reply test/sql/test_array_fn/R/test_array_cast:53]: done

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 14 / 16 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|4|6|66.67%|[1126, 1127]|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 14 / 16 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|4|6|66.67%|[1126, 1127]|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 14 / 16 (87.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Type.java|4|6|66.67%|[1126, 1127]|
|:large_blue_circle:|com/starrocks/catalog/FunctionSet.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ExpressionAnalyzer.java|6|6|100.00%|[]|

**Pull request closed without merging**